### PR TITLE
Add test for Transaction Payment on zero balance account

### PR DESCRIPTION
**Pull request by shawntabrizi**
This adds a test which confirms that an account with zero balance, triggering a transaction with `Pays::No`, will not trigger any events such as `AccountKilled`

Closes: #6159

### CODE REVIEW

**Comment by shawntabrizi** [comment]
Found that this test does not actually reveal any issues, but seems like a good enough test to add anyway.

File: frame/transaction-payment/src/lib.rs [hunk 1]
```diff
@@ -857,4 +874,36 @@ mod tests {
 			assert_eq!(Balances::free_balance(2), 200 - 5 - 10 - 100 - 5);
 		});
 	}
+
+	#[test]
+	fn zero_transfer_on_zero_account() {
```
Comment (bkchr) [inline frame/transaction-payment/src/lib.rs:879]:
```suggestion
	fn zero_transfer_on_free_transaction() {
```
